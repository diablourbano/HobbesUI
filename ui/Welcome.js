import React from 'react';
import { DrawerActions } from 'react-navigation-drawer';
import * as styled from './styles';

export const Welcome = (props) => {
  const { navigation } = props;

  return (
    <styled.MainContainer>
      <styled.Content>
        <styled.MainSection>
          <styled.Title>hello</styled.Title>
          <styled.Subtitle>Styleguide System</styled.Subtitle>
        </styled.MainSection>
        <styled.Footer>
          <styled.Note
            onPress={() => {
              navigation.dispatch(DrawerActions.toggleDrawer());
            }}
          >
            open the drawer or slide :)
          </styled.Note>
        </styled.Footer>
      </styled.Content>
    </styled.MainContainer>
  );
}
