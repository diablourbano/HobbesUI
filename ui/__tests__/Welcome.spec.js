import React from 'react';
import { shallow } from 'enzyme';
import { Welcome } from '../Welcome';

describe('Welcome screen', () => {
  it('should match snapshot', () => {
    const component = shallow(<Welcome />);
    expect(component).toMatchSnapshot();
  });
});
