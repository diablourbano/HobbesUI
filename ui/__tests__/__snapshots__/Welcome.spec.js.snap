// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome screen should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Welcome />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <styles__Content>
        <styles__MainSection>
          <styles__Title>
            hello
          </styles__Title>
          <styles__Subtitle>
            Styleguide System
          </styles__Subtitle>
        </styles__MainSection>
        <styles__Footer>
          <styles__Note
            onPress={[Function]}
          >
            open the drawer or slide :)
          </styles__Note>
        </styles__Footer>
      </styles__Content>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styles__MainSection>
            <styles__Title>
              hello
            </styles__Title>
            <styles__Subtitle>
              Styleguide System
            </styles__Subtitle>
          </styles__MainSection>,
          <styles__Footer>
            <styles__Note
              onPress={[Function]}
            >
              open the drawer or slide :)
            </styles__Note>
          </styles__Footer>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styles__Title>
                hello
              </styles__Title>,
              <styles__Subtitle>
                Styleguide System
              </styles__Subtitle>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "hello",
              },
              "ref": null,
              "rendered": "hello",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Styleguide System",
              },
              "ref": null,
              "rendered": "Styleguide System",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <styles__Note
              onPress={[Function]}
            >
              open the drawer or slide :)
            </styles__Note>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "open the drawer or slide :)",
              "onPress": [Function],
            },
            "ref": null,
            "rendered": "open the drawer or slide :)",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <styles__Content>
          <styles__MainSection>
            <styles__Title>
              hello
            </styles__Title>
            <styles__Subtitle>
              Styleguide System
            </styles__Subtitle>
          </styles__MainSection>
          <styles__Footer>
            <styles__Note
              onPress={[Function]}
            >
              open the drawer or slide :)
            </styles__Note>
          </styles__Footer>
        </styles__Content>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styles__MainSection>
              <styles__Title>
                hello
              </styles__Title>
              <styles__Subtitle>
                Styleguide System
              </styles__Subtitle>
            </styles__MainSection>,
            <styles__Footer>
              <styles__Note
                onPress={[Function]}
              >
                open the drawer or slide :)
              </styles__Note>
            </styles__Footer>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styles__Title>
                  hello
                </styles__Title>,
                <styles__Subtitle>
                  Styleguide System
                </styles__Subtitle>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "hello",
                },
                "ref": null,
                "rendered": "hello",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Styleguide System",
                },
                "ref": null,
                "rendered": "Styleguide System",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <styles__Note
                onPress={[Function]}
              >
                open the drawer or slide :)
              </styles__Note>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "open the drawer or slide :)",
                "onPress": [Function],
              },
              "ref": null,
              "rendered": "open the drawer or slide :)",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
